src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    private static class Node<K, V> {
+        private final K key;
+        private Node<K, V> next;
+        private V value;
+
+        private Node(V value, K key, Node<K, V> next) {
+            this.value = value;
+            this.key = key;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || threshold == size) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(value, key, null);
+        if (table[getIndex(key)] == null) {
+            table[getIndex(key)] = newNode;
+            size++;
+        } else {
+            Node<K, V> currentNode = table[getIndex(key)];
+            while (currentNode.next != null && !Objects.equals(currentNode.key, key)) {
+                currentNode = currentNode.next;
+            }
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+            } else {
+                currentNode.next = newNode;
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> currentNode = findCurrentNode(key);
+        return currentNode != null ? currentNode.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(Object key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        if (table == null) {
+            table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+            threshold = (int)(DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+        } else {
+            Node<K, V>[] oldArray = table;
+            table = (Node<K, V>[]) new Node[oldArray.length << 1];
+            threshold = (int)(table.length * DEFAULT_LOAD_FACTOR);
+            rePlaceNode(oldArray);
+        }
+    }
+
+    private Node<K, V> findCurrentNode(K key) {
+        if (size == 0) {
+            return null;
+        }
+        Node<K, V> currentNode = table[getIndex(key)];
+        while (!Objects.equals(currentNode.key, key) && currentNode.next != null) {
+            currentNode = currentNode.next;
+        }
+        return Objects.equals(currentNode.key, key) ? currentNode : null;
+    }
+
+    private void rePlaceNode(Node<K, V>[] oldArray) {
+        size = 0;
+        for (Node<K, V> element : oldArray) {
+            if (element != null) {
+                put(element.key, element.value);
+                Node<K, V> currentNode = element;
+                while (currentNode.next != null) {
```

Comment:
Do you need 2 'not null' check here and currecntNode.next?
The 'foreach' loop runs on all nodes.
Collection impliment the interface [Iterable](https://mate.academy/learn/java-core/java-array-list/practice?section=theory).